src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && this[i] === undefined) {
+        break;
+      }
+
+      if (i === this.length - 1) {
+        joinedString += this[i];
+        break;
+      }
+
+      if (this[i] === null || this[i] === undefined) {
+        joinedString += separator;
+        continue;
+      }
```

Comment:
why don't you just add `this[i]` to string if it's NOT `null` or `undefined`, you're repeating yourself in next lines - try to invert your if statement to do that.